hudi-utilities/src/test/java/org/apache/hudi/utilities/sources/helpers/TestProtoConversionUtil.java

```diff
@@ -130,10 +141,43 @@ public void recursiveSchema_withOverflow() throws Exception {
     Assertions.assertEquals(input.getChildren(1).getRecurseField().getRecurseField(), parsedChildren2Overflow);
   }
 
+  @Test
+  public void oneOfSchema() throws IOException {
+    Schema.Parser parser = new Schema.Parser();
+    Schema convertedSchema = parser.parse(getClass().getClassLoader().getResourceAsStream("schema-provider/proto/oneof_schema.avsc"));
+    WithOneOf input = WithOneOf.newBuilder().setLong(32L).build();
+    GenericRecord actual = serializeAndDeserializeAvro(ProtoConversionUtil.convertToAvro(convertedSchema, input), convertedSchema);
+
+    GenericData.Record expectedRecord = new GenericData.Record(convertedSchema);
+    expectedRecord.put("int", null);
+    expectedRecord.put("long", 32L);
+    expectedRecord.put("message", null);
+    Assertions.assertEquals(expectedRecord, actual);
+  }
+
+  private void assertUnsignedLongCorrectness(Schema convertedSchema, Sample input, GenericRecord actual, boolean wellKnownTypesAsRecords) {
```

Comment:
Good indicator of whether method is coherent is how easy it to judge what it does by simply looking at its signature. For this one it's not that easy.

I'd suggest to split it actually in 2 and instead make them like following:

```
assert(Schema.Field fieldSchema, Long expectedValue, ... actual, ...)
```